PWGCF/EbyEFluctuations/Tasks/EbyEmeanPtFluc.cxx

```diff
@@ -0,0 +1,777 @@
+      hist.fill(HIST("Analysis/Pion/h_mean_Q1_Mult_pi_tpc"), NTPC, mean_Q1_Pi_tpc, N_FT0M);
+      hist.fill(HIST("Analysis/Pion/h_twopart_Mult_pi_tpc"), NTPC, twopart_Pi_tpc, N_FT0M);
+    }
+    if (N_Ka_tpc > 1) {
+      double mean_Q1_Ka_tpc = Q1_Ka_tpc / static_cast<double>(N_Ka_tpc);
+      double twopart_Ka_tpc = ((Q1_Ka_tpc * Q1_Ka_tpc) - Q2_Ka_tpc) / (static_cast<double>(N_Ka_tpc) * (static_cast<double>(N_Ka_tpc) - 1));
+      hist.fill(HIST("Analysis/Kaon/h_mean_Q1_Mult_ka_tpc"), NTPC, mean_Q1_Ka_tpc, N_FT0M);
+      hist.fill(HIST("Analysis/Kaon/h_twopart_Mult_ka_tpc"), NTPC, twopart_Ka_tpc, N_FT0M);
+    }
+    if (N_Pr_tpc > 1) {
+      double mean_Q1_Pr_tpc = Q1_Pr_tpc / static_cast<double>(N_Pr_tpc);
+      double twopart_Pr_tpc = ((Q1_Pr_tpc * Q1_Pr_tpc) - Q2_Pr_tpc) / (static_cast<double>(N_Pr_tpc) * (static_cast<double>(N_Pr_tpc) - 1));
+      hist.fill(HIST("Analysis/Proton/h_mean_Q1_Mult_pr_tpc"), NTPC, mean_Q1_Pr_tpc, N_FT0M);
+      hist.fill(HIST("Analysis/Proton/h_twopart_Mult_pr_tpc"), NTPC, twopart_Pr_tpc, N_FT0M);
+    }
+
+    //-----------------------TPC + TOF (No p cuts)--------------------------//
+    if (Nch_tof > 1) {
+      double mean_Q1_tof = Q1_tof / static_cast<double>(Nch_tof);
+      double twopart_tof = ((Q1_tof * Q1_tof) - Q2_tof) / (static_cast<double>(Nch_tof) * (static_cast<double>(Nch_tof) - 1));
+      hist.fill(HIST("Analysis/Charged/h_mean_Q1_Mult_ch_tof"), NTPC, mean_Q1_tof, N_FT0M);
+      hist.fill(HIST("Analysis/Charged/h_twopart_Mult_ch_tof"), NTPC, twopart_tof, N_FT0M);
+    }
+    if (N_Pi_tof > 1) {
+      double mean_Q1_Pi_tof = Q1_Pi_tof / static_cast<double>(N_Pi_tof);
+      double twopart_Pi_tof = ((Q1_Pi_tof * Q1_Pi_tof) - Q2_Pi_tof) / (static_cast<double>(N_Pi_tof) * (static_cast<double>(N_Pi_tof) - 1));
+      hist.fill(HIST("Analysis/Pion/h_mean_Q1_Mult_pi_tof"), NTPC, mean_Q1_Pi_tof, N_FT0M);
+      hist.fill(HIST("Analysis/Pion/h_twopart_Mult_pi_tof"), NTPC, twopart_Pi_tof, N_FT0M);
+    }
+    if (N_Ka_tof > 1) {
+      double mean_Q1_Ka_tof = Q1_Ka_tof / static_cast<double>(N_Ka_tof);
+      double twopart_Ka_tof = ((Q1_Ka_tof * Q1_Ka_tof) - Q2_Ka_tof) / (static_cast<double>(N_Ka_tof) * (static_cast<double>(N_Ka_tof) - 1));
+      hist.fill(HIST("Analysis/Kaon/h_mean_Q1_Mult_ka_tof"), NTPC, mean_Q1_Ka_tof, N_FT0M);
+      hist.fill(HIST("Analysis/Kaon/h_twopart_Mult_ka_tof"), NTPC, twopart_Ka_tof, N_FT0M);
+    }
+    if (N_Pr_tof > 1) {
+      double mean_Q1_Pr_tof = Q1_Pr_tof / static_cast<double>(N_Pr_tof);
+      double twopart_Pr_tof = ((Q1_Pr_tof * Q1_Pr_tof) - Q2_Pr_tof) / (static_cast<double>(N_Pr_tof) * (static_cast<double>(N_Pr_tof) - 1));
+      hist.fill(HIST("Analysis/Proton/h_mean_Q1_Mult_pr_tof"), NTPC, mean_Q1_Pr_tof, N_FT0M);
+      hist.fill(HIST("Analysis/Proton/h_twopart_Mult_pr_tof"), NTPC, twopart_Pr_tof, N_FT0M);
+    }
+  }
```

Comment:
In the `process` function the code is basically replicated for the different track cases and ID scenarios
Replicating code is quite quite error prone and complicated to debug it
I would suggest trying to unify the code as much as possible being in simple (templated) routines or lambda functions

Please, take this comment as a suggestion